tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java

```diff
@@ -1,7 +1,34 @@
 package edu.utexas.tacc.tapis.security.api.requestBody;
 
-public class ReqAddChildRole 
+import org.apache.commons.lang3.StringUtils;
+
+import edu.utexas.tacc.tapis.shared.i18n.MsgUtils;
+
+public class ReqAddChildRole
+ implements IReqBody
 {
+    public String tenant;
+    public String user;
     public String parentRoleName;
     public String childRoleName;
+    
+    /** Return a user-appropriate error message on failed validation
+     *  and return null if validation succeeds.
+     */ 
+    @Override
+    public String validate() 
```

Comment:
Any reason not to use the javax.validations @NotNull annotations?

Reply:
Annotations an alternative way of doing things.  One of the downsides is that checking for empty strings is almost always also required.  I'm also not sure what type of error messages get generated using the annotation.  It's worth considering.

Reply:
There is also @NotEmpty and @NotBlank